dashboard/app/models/experiments/pilot.rb

```diff
@@ -14,4 +14,10 @@
 #  index_pilots_on_name  (name)
 #
 class Pilot < ApplicationRecord
+  # The name can be used in urls, constrain it to all lowercase with no spaces
+  validates :name, presence: true, format: {with: /\A[a-z0-9-]+\z/}
```

Comment:
I think you can just validate uniqueness of name here instead of worrying about a follow-up database migration, I've heard mixed things about whether it's preferable to just enforce stuff like that at the app level vs. db level? Curious on your thoughts?

https://apidock.com/rails/ActiveRecord/Validations/ClassMethods/validates_uniqueness_of

Reply:
Goooooood question.  :-)  There's a range of opinions on this and there are pros and cons to each approach.  I view validations in the rails model as the things you want to enforce when creating a new item or editing an existing one (but they don't guarantee anything about data that was written in the past).  Constraints in the database are things you want to be sure is true for all your data in the database, regardless of when/how it was added.

For validating uniqueness, using rails validation is extra suspect because it's vulnerable to race conditions.

I'm curious as to what our general philosophy on this at Code.org is.